### PR TITLE
CloudWatch: Allow use of missing AWS namespaces using custom metrics

### LINKED ISSUE #22215: AWS CloudWatch: Custom Metrics namespaces cannot be used to add missing AWS Namespaces
AWS Namespaces & their dimensions for the Cloudwatch Datasource are hardcoded here:
https://github.com/grafana/grafana/blob/master/pkg/tsdb/cloudwatch/metric_find_query.go#L42
https://github.com/grafana/grafana/blob/master/pkg/tsdb/cloudwatch/metric_find_query.go#L125

There's a gap right now and some Namespaces are not defined, e.g..
- AWS/Athena
- AWS/Cassandra
- AWS/DataSync
- AWS/ElastiCache
- AWS/FSx
- AWS/QLDB
- AWS/Robomaker
- AWS/SDKMetrics
- AWS/ServiceCatalog
- AWS/SSM-RunCommand
- AWS/Transfer
- AWS/WorkMail

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html

When I tried to include some of these using the "Custom Metrics namespace" setting of the Datasource, it turned out that any AWS Namespace configured there is not usable due to this function:
```
func isCustomMetrics(namespace string) bool {
	return strings.Index(namespace, "AWS/") != 0
}
```
https://github.com/grafana/grafana/blob/master/pkg/tsdb/cloudwatch/metric_find_query.go#L838-L840
when it returns `true`, the Metrics & Dimensions won't be fetched from AWS, but will just be looked up in the `metricsMap` / `dimensionsMap`
I suggest the `isCustomMetrics` function should always return `true` when the namespace does not exist in the `metricsMap` / `dimensionsMap`

Right now in the end, you can select the custom AWS Namespace in Grafana, but the Metric dropdown will stay empty.

(This is related to https://github.com/grafana/grafana/pull/22179)

### PR DESCRIPTION
**Pull request by mmcoltman**
**What this PR does / why we need it**:

This lets CloudWatch namespaces provided by AWS but which are missing from the Grafana project be added to the datasource as a Custom Metric namespace.

**Which issue(s) this PR fixes**:

Fixes #22215

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tsdb/cloudwatch/metric_find_query_test.go
Comment (mmcoltman) [inline pkg/tsdb/cloudwatch/metric_find_query_test.go:523]: This failed before the change to isCustomMetrics().

**Comment by sunker** [comment]
Thanks for contributing @mmcoltman! 

We're considering introducing a new way of handling custom namespaces. Please see [this](https://github.com/grafana/grafana/issues/30977) issue. Even though this looks like a nice fix, we won't need it if custom namespaces are loaded from the ListMetrics API. Therefore I don't want to merge this PR until we know when #30977 will be implemented.

**Comment by sunker** [comment]
@mtanda would be great if you could take a peek at this since you've worked with custom namespaces before.

**Comment by mtanda** [comment]
Looks good to me. Thanks for contributing!

**Review by mtanda — APPROVED** [review]

**Review by sunker — APPROVED** [review]
LGTM

**Comment by sunker** [comment]
@mmcoltman do you want to resolve the conflicts so that we can merge? thanks!

**Comment by mmcoltman** [comment]
> @mmcoltman do you want to resolve the conflicts so that we can merge? thanks!

@sunker Sure, done now.

**Comment by sunker** [comment]
Thanks @mmcoltman! You also need to sign the CLA before I can merge.

**Comment by mmcoltman** [comment]
> Thanks @mmcoltman! You also need to sign the CLA before I can merge.

@sunker Ah, I assumed it was a one-time thing. That should also be done now, thanks.

**Review by sunker — APPROVED** [review]

**Review by sunker — APPROVED** [review]

**Comment by sunker** [comment]
@mmcoltman do you want to sync this branch with the main branch? Sorry for the delay. 
Let's aim to get this into 8.0 beta2 which is due next week.

**Comment by mmcoltman** [comment]
> @mmcoltman do you want to sync this branch with the main branch? Sorry for the delay.
> Let's aim to get this into 8.0 beta2 which is due next week.

No worries, should done now!

**Pull request merged**
